rfcs/CompositeSchemas.md

```diff
@@ -0,0 +1,26 @@
+# RFC: GraphQL Composite Schemas
+
+There's a _lot_ of different ways of combining multiple GraphQL schemas together to form a larger composite schema, and the wide variety of options can make it challenging to design schemas in a way that will be easy to compose later. To name but a few approaches:
+
+* Schema merging and stitching (various implementations in many languages)
+* GraphQL modules and other "extend type"-based approaches
+* Federation (Apollo's v1 and v2, Mercurius, WunderGraph, Hot Chocolate, etc)
+* Hasura's GraphQL Joins
+
+Though these are all separate solutions to similar problems, there are various concerns that most of them have to consider:
+
+* Identifying the schemas to join
+* Discovering/defining the relationships between the types/schemas
+* Schema manipulation to avoid conflicts (e.g. renaming types/fields)
+* Declaring which schemas "own" a particular type, and detecting conflicts
```

Comment:
I think we should remove the use of the term “ownership” from the list of concerns as type ownership is a potential design pattern/anti-pattern depending on your point of view.

we can change this to a general call to conflict prevention/detection, which should go above the schema manipulation item directly above

Reply:
I agree with this, something I miss in current implementations is field redundancy within multiple sub/schemas. E.g. multiple sub-schemas could implement a specific field and the gateway could then resolve the field on either of those sub-schemas. This is also something that would migrating a field from on sub-schema to another easier.

Reply:
In terms of another concept related to ownership without endorsement of the pattern, we could add a final concern: “managing composition metadata with respect to handling the above concerns”

for instance, you may think Federation 1 type ownership is helpful because it obviates the need to annotate which schema’s type description is selected for shared types, versus @gmac ‘s introduction of canonical directive within schema stitching where the composition metadata is made explicit rather than flowing from the ownership model….

indeed, one of the immediate goals of this working group might be to consolidate around a set of directive names that should be reserved to represent composition metadata, which could happen well before the complete specification of the directives…

if there are other advantages of “ownership” not covered by the remaining categories that anyone can think of, I would suggest posting here and adding to the list

Reply:
Either way, "ownership" is a shared concern, whether the individual solutions decide that one schema owns the type versus shared ownership across all schemas.

Reply:
Solutions without ownership have to carefully manage redundancy, which is a shared concern, but don’t have to concern themselves with ownership at all because shared ownership is kind of meaningless.

The term itself is nebulous and invites more questions than answers.

Not a huge deal, of course, but this is (imo) the solution space slipping into the description of the problem and should be avoided.

Reply:
Added based on feedback :+1: